ci/cirrus.sh

```diff
@@ -16,10 +15,24 @@ print_environment() {
             EXPERIMENTAL ECDH RECOVERY SCHNORRSIG \
             SECP256K1_TEST_ITERS BENCH SECP256K1_BENCH_ITERS CTIMETEST\
             EXAMPLES \
-            WRAPPER_CMD CC AR NM HOST
+            HOST WRAPPER_CMD
     do
         eval 'printf "%s %s " "$i=\"${'"$i"'}\""'
     done
+    # In contrast to the previous variables, CC, CFLAGS, AR and NM may be unset.
+    # Print them if they are set (even if set to the empty string).
+    if [ -n "${CC+x}" ]; then
+        echo -n "CC=\"${CC}\" "
+    fi
+    if [ -n "${CFLAGS+x}" ]; then
+        echo -n "CFLAGS=\"${CFLAGS}\" "
+    fi
+    if [ -n "${AR+x}" ]; then
+        echo -n "AR=\"${AR}\" "
+    fi
+    if [ -n "${NM+x}" ]; then
+        echo -n "NM=\"${NM}\" "
+    fi
```

Comment:
for loop and without `echo -n` which is not POSIX (I don't think we really care in CI but so far we tried to stick to POSIX) 
```suggestion
    for var in CC CFLAGS AR NM; do
        eval "isset=\${$var+x}"
        if [ -n "$isset" ]; then
            eval "val=\${$var}"
            # shellcheck disable=SC2154
            printf '%s="%s" ' "$var" "$val"
        fi
    done
```

Reply:
Can't find the point where `shellcheck` is called... Did I miss a script in CI?

Reply:
With the suggested loop, why do separate CC, CFLAGS, AR and NM  variables from others? A tiny overhead for a check seems an acceptable tradeoff for the succinct code, no?

Reply:
> With the suggested loop, why do separate CC, CFLAGS, AR and NM variables from others? A tiny overhead for a check seems an acceptable tradeoff for the succinct code, no?

Oh indeed. If we have a working loop that handles arbitrary variables, we should use it for all variables. Actually this makes me more convinced of my suggestion than I was before (the eval is somehwat ugly but having one piece of code for all env variables is probably a good idea.)  @jonasnick What do you think?  

> Can't find the point where shellcheck is called... Did I miss a script in CI?

Sorry, I called it locally on my machine. :D I've seen it being very useful a few times in Core, and so I thought it doesn't hurt to add the shellcheck comment here (even if maintainers use locally). Perhaps we want to add it in the future. If @jonasnick picks up this suggestion at all, I'll let him decide if the comment is appropriate / consistent without running shellcheck in CI. ;)

Reply:
> I've seen it being very useful a few times in Core,

Indeed.

Reply:
Thanks @real-or-random, with your snippet I was able to merge these variables with the loop above. I'd keep the shellcheck directive, since I'm using shellcheck locally.